sanic/cookies/response.py

```diff
@@ -0,0 +1,617 @@
+        elif secure_prefix:
+            if not secure:
+                raise ServerError(
+                    "Cannot set secure_prefix on a cookie without secure=True"
+                )
+            key = self.SECURE_PREFIX + key
+        if partitioned and not host_prefix:
+            # This is technically possible, but it is not advisable so we will
+            # take a stand and say "don't shoot yourself in the foot"
+            raise ServerError(
+                "Cannot create a partitioned cookie without "
+                "also setting host_prefix=True"
+            )
+
+        self.key = key
+        self.value = value
+        super().__init__()
+        self._set_value("path", path)
+        self._set_value("expires", expires)
+        self._set_value("comment", comment)
+        self._set_value("domain", domain)
+        self._set_value("max-age", max_age)
+        self._set_value("secure", secure)
+        self._set_value("httponly", httponly)
+        self._set_value("samesite", samesite)
+        self._set_value("partitioned", partitioned)
+
+    def __setitem__(self, key, value):
+        deprecation(
+            "Setting values on a Cookie object as a dict has been deprecated. "
+            "This feature will be removed in v24.3. You should instead set "
+            f"values on cookies as object properties: cookie.{key}=... ",
+            24.3,
+        )
+        self._set_value(key, value)
+
+    # This is a temporary method for backwards compat and should be removed
+    # in v24.3 when this is no longer a dict
+    def _set_value(self, key: str, value: Any) -> None:
+        if key not in self._keys:
+            raise KeyError("Unknown cookie property: %s=%s" % (key, value))
+
+        if value is not None:
+            if key.lower() == "max-age" and not str(value).isdigit():
+                raise ValueError("Cookie max-age must be an integer")
+            elif key.lower() == "expires" and not isinstance(value, datetime):
+                raise TypeError("Cookie 'expires' property must be a datetime")
+            elif key.lower() == "samesite":
+                if value.lower() not in SAMESITE_VALUES:
+                    raise TypeError(
+                        "Cookie 'samesite' property must "
+                        f"be one of: {','.join(SAMESITE_VALUES)}"
+                    )
+                value = value.title()
+
+        super().__setitem__(key, value)
+
+    def encode(self, encoding):
+        """
+        Encode the cookie content in a specific type of encoding instructed
+        by the developer. Leverages the :func:`str.encode` method provided
+        by python.
+
+        This method can be used to encode and embed ``utf-8`` content into
+        the cookies.
+
+        :param encoding: Encoding to be used with the cookie
+        :return: Cookie encoded in a codec of choosing.
+        :except: UnicodeEncodeError
+        """
+        return str(self).encode(encoding)
```

Comment:
I believe this method is not used by Sanic (which instead converts to str and then encodes). Could you check if this can be removed? Notably it also does not support the `errors` kwarg that we use on `encode`.

Reply:
We can deprecate, not remove.

I agree, we are not using it in Sanic.

```py
    @property
    def processed_headers(self) -> Iterator[Tuple[bytes, bytes]]:
		...
        return (
            (name.encode("ascii"), f"{value}".encode(errors="surrogateescape"))
            for name, value in self.headers.items()
        )
```

Reply:
I am not sure I agree with deprecation of this though. Presumably you could be using this to manually encode cookies, which we otherwise would not support.